### PR TITLE
Add Resource Dictionary Template

### LINKED ISSUE #4534: [Templates] No ResourceDictionary template
### Description

Please can we have a ResourceDictionary item template in MAUI? It's been raised what feels like a million times now over the last X years. Everyone seems to agree it's a good idea and that it should happen, but it never does.

It's a source of frustration for users for years that to add an RD you have to add a ContentPage or ContentView then change the base class in XAML and code-behind.

### Steps to Reproduce

gif

### Version with bug

Preview 7

### Last version that worked well

Preview 8

### Affected platforms

iOS

### Affected platform versions

gif

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change

Adds an empty `ResourceDictionary` to the templates to make adding them less of a pain

### Issues Fixed

Fixes #4534

### CODE REVIEW

File: src/Templates/src/templates/maui-resourcedictionary-xaml/.template.config/ide.host.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "$schema": "http://json.schemastore.org/vs-2017.3.host",
+    "appliesTo": "UseMaui",
+    "defaultItemExtension": "xaml",
+    "itemHierarchyPaths": [ ".NET MAUI" ],
+    "icon": "ide/MauiContentViewXamlFile.ico",
```
Comment (jsuarezruiz) [inline src/Templates/src/templates/maui-resourcedictionary-xaml/.template.config/ide.host.json:6]: Could we use a different icon?

File: src/Templates/src/templates/maui-resourcedictionary-xaml/NewContent1.xaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ResourceDictionary xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="MauiApp1.NewContent1">
```
Comment (jsuarezruiz) [inline src/Templates/src/templates/maui-resourcedictionary-xaml/NewContent1.xaml:4]: Maybe could be nice to call it **Resource1** to have the same experience as before with Resource Dictionaries in other frameworks 
image

Comment (jfversluis) [reply src/Templates/src/templates/maui-resourcedictionary-xaml/NewContent1.xaml:4]: You mean Resource1 or Dictionary1?

File: src/Templates/src/templates/maui-resourcedictionary-xaml/.template.config/ide.host.json (hunk 1 above)
Comment (jfversluis) [reply src/Templates/src/templates/maui-resourcedictionary-xaml/.template.config/ide.host.json:6]: Very good! Any clue where we get that? :D

File: src/Templates/src/templates/maui-resourcedictionary-xaml/NewContent1.xaml (hunk 2 above)
Comment (jsuarezruiz) [reply src/Templates/src/templates/maui-resourcedictionary-xaml/NewContent1.xaml:4]: Dictionary1

**Comment by jfversluis** [comment]
Future us: need to get the localization for this one in as well after #5352 is merged

**Review by Redth — APPROVED** [review]

**Pull request merged**